src/app/modules/payments/stripe.controller.ts

```diff
@@ -191,17 +193,70 @@ export const getPaymentReceipt: ControllerHandler<{
       logger.info({
         message: 'Received receipt url from Stripe webhook',
         meta: {
-          action: 'getPaymentReceipt',
+          action: 'checkPaymentReceiptStatus',
           receiptUrl,
         },
       })
-      res.status(StatusCodes.OK).send({ receipt: receiptUrl })
+
+      return res.status(StatusCodes.OK).json({ isReady: true })
     })
     .mapErr((error) => {
       logger.error({
         message: 'Error retrieving receipt URL',
         meta: {
-          action: 'getPaymentReceipt',
+          action: 'checkPaymentReceiptStatus',
+          formId: formId,
+          submissionId: submissionId,
+        },
+        error,
+      })
+      return res.status(StatusCodes.NOT_FOUND).json({ message: error })
+    })
+}
+
+export const downloadPaymentReceipt: ControllerHandler<{
+  formId: string
+  submissionId: string
+}> = (req, res) => {
+  const { formId, submissionId } = req.params
+  logger.info({
+    message: 'downloadPaymentReceipt endpoint called',
+    meta: {
+      action: 'downloadPaymentReceipt',
+      formId: formId,
+      submissionId: submissionId,
+    },
+  })
+
+  return StripeService.getReceiptURL(formId, submissionId)
+    .map((receiptUrl) => {
+      logger.info({
+        message: 'Received receipt url from Stripe webhook',
+        meta: {
+          action: 'downloadPaymentReceipt',
+          receiptUrl,
+        },
+      })
+      // retrieve receiptURL as html
+      return (
+        axios
+          .get<string>(receiptUrl)
+          // convert to pdf and return
+          .then((receiptUrlResponse) => {
+            const html = receiptUrlResponse.data
+            const pdfBufferPromise = generatePdfFromHtml(html)
+            return pdfBufferPromise
+          })
+          .then((pdfBuffer) => {
+            return res.status(StatusCodes.OK).json({ pdfBuffer })
```

Comment:
I don't think json is the right medium to send a full file content. Let's have this endpoint return the file as-is, And supply relevant headers:
* [Content-Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)
* [Content-Disposition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition)

Also [Content-Length](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length) if express doesn't include that by default.

Reply:
[9a2c271](https://github.com/opengovsg/FormSG/pull/5837/commits/9a2c2716692a8899b0f75ce1f9bd3f0fca1cc002)

Content-Length is automatically assigned by express-js ([source](http://expressjs.com/en/4x/api.html))